Snail/Disposer.swift

```diff
@@ -8,15 +8,24 @@ public protocol DisposableType {
 
 public class Disposer {
     private(set) public var disposables: [DisposableType] = []
+    private let disposablesQueue = DispatchQueue(label: "snail-disposer-queue", attributes: .concurrent)
 
     public init() {}
 
+    deinit {
+        disposeAll()
+    }
+
     public func disposeAll() {
-        disposables.forEach { $0.dispose() }
-        disposables.removeAll()
+        disposablesQueue.sync {
+            self.disposables.forEach { $0.dispose() }
+            self.disposables.removeAll()
+        }
```

Comment:
If using a concurrent queue i think this should be:
```suggestion
        disposablesQueue.sync { self.disposables.forEach { $0.dispose } }
        disposablesQueue.async(flags: .barrier) { self.disposables.removeAll() }
```

Checkout: https://www.avanderlee.com/swift/concurrent-serial-dispatchqueue/ for more details.